### PR TITLE
Fix #11571 part 1 - create gfx_draw_string_left() overload

### LINKED ISSUE #11570: Use ScreenCoordsXY on gfx_draw_string_centred
Create an overload of `gfx_draw_string_centred` on `\src\openrct2\drawing\Text.cpp` with the same name, but that takes a `const ScreenCoordsXY&` object instead of an `int32_t x` and and an `int32_t y` and replace the calls to the `{x ,y}` pair by the one using `ScreenCoordsXY`.

Don't try to do all calls at once, unless it doesn't occur a lot, as this might lead to a huge PR. When we stop using the old function, we can drop it entirely in favour of the `ScreenCoordsXY` alternative.

### LINKED ISSUE #11571: Use ScreenCoordsXY on gfx_draw_string_left
Create an overload of `gfx_draw_string_left` on `\src\openrct2\drawing\Text.cpp` with the same name, but that takes a `const ScreenCoordsXY&` object instead of an `int32_t x` and and an `int32_t y` and replace the calls to the `{x ,y}` pair by the one using `ScreenCoordsXY`.

Don't try to do all calls at once, unless it doesn't occur a lot, as this might lead to a huge PR. When we stop using the old function, we can drop it entirely in favour of the `ScreenCoordsXY` alternative.

### PR DESCRIPTION
**Pull request by TomasZilinek**
I created the overload, and replaced about a thrid of all occurences of the old function since there are too many calls to it.

### CODE REVIEW

File: src/openrct2-ui/windows/Cheats.cpp [hunk 1]
```diff
@@ -1207,14 +1207,14 @@ static void window_cheats_paint(rct_window* w, rct_drawpixelinfo* dpi)
         }
         int32_t actual_month = _monthSpinnerValue - 1;
         gfx_draw_string_left(
-            dpi, STR_BOTTOM_TOOLBAR_CASH, gCommonFormatArgs, colour, w->windowPos.x + XPL(0) + TXTO,
-            w->windowPos.y + YPL(2) + TXTO);
+            dpi, STR_BOTTOM_TOOLBAR_CASH, gCommonFormatArgs, colour,
+            { w->windowPos.x + XPL(0) + TXTO, w->windowPos.y + YPL(2) + TXTO });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Cheats.cpp:1211]: When one of the objects is `ScreenCoordsXY` already, we're sticking to the following convention:
```suggestion
            w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(2) + TXTO });
```

This happens in other places in the PR as well, but I'll not open issues for all, please make sure to review the whole diff.

File: src/openrct2-ui/windows/EditorInventionsList.cpp [hunk 2]
```diff
@@ -815,12 +815,10 @@ static void window_editor_inventions_list_drag_moved(rct_window* w, const Screen
 static void window_editor_inventions_list_drag_paint(rct_window* w, rct_drawpixelinfo* dpi)
 {
     rct_string_id drawString;
-    int32_t x, y;
+    auto screenCoords = ScreenCoordsXY{ w->windowPos.x, w->windowPos.y + 2 };
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorInventionsList.cpp:818]:
```suggestion
    auto screenCoords = w->windowPos + ScreenCoordsXY{ 0, 2 };
```

File: src/openrct2-ui/windows/Cheats.cpp [hunk 3]
```diff
@@ -1207,55 +1207,55 @@ static void window_cheats_paint(rct_window* w, rct_drawpixelinfo* dpi)
...
-            dpi, STR_DAY, nullptr, COLOUR_BLACK, w->windowPos.x + XPL(0) + TXTO, w->windowPos.y + YPL(9) + TXTO);
+            dpi, STR_DAY, nullptr, COLOUR_BLACK, w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(9) + TXTO });
         gfx_draw_string_right(
             dpi, STR_FORMAT_INTEGER, &_yearSpinnerValue, w->colours[1],
-            { w->windowPos.x + WPL(1) - 34 - TXTO, w->windowPos.y + YPL(7) + TXTO });
+            w->windowPos + ScreenCoordsXY{ WPL(1) - 34 - TXTO, YPL(7) + TXTO });
         gfx_draw_string_right(
             dpi, STR_FORMAT_MONTH, &actual_month, w->colours[1],
-            { w->windowPos.x + WPL(1) - 34 - TXTO, w->windowPos.y + YPL(8) + TXTO });
+            w->windowPos + ScreenCoordsXY{ WPL(1) - 34 - TXTO, YPL(7) + TXTO });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Cheats.cpp:1223]:
```suggestion
            w->windowPos + ScreenCoordsXY{ WPL(1) - 34 - TXTO, YPL(8) + TXTO });
```
Used to be `8`

File: src/openrct2-ui/windows/Cheats.cpp [hunk 4]
```diff
@@ -1207,55 +1207,55 @@ static void window_cheats_paint(rct_window* w, rct_drawpixelinfo* dpi)
...
         gfx_draw_string_right(
             dpi, STR_FORMAT_INTEGER, &_daySpinnerValue, w->colours[1],
-            { w->windowPos.x + WPL(1) - 34 - TXTO, w->windowPos.y + YPL(9) + TXTO });
+            w->windowPos + ScreenCoordsXY{ WPL(1) - 34 - TXTO, YPL(7) + TXTO });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Cheats.cpp:1226]:
```suggestion
            w->windowPos + ScreenCoordsXY{ WPL(1) - 34 - TXTO, YPL(9) + TXTO });
```

File: src/openrct2-ui/windows/Cheats.cpp [hunk 5]
```diff
@@ -1207,55 +1207,55 @@ static void window_cheats_paint(rct_window* w, rct_drawpixelinfo* dpi)
...
-            dpi, STR_CHEAT_GUEST_THIRST, nullptr, COLOUR_BLACK, w->windowPos.x + XPL(0) + TXTO, w->windowPos.y + YPL(4) + TXTO);
+            dpi, STR_CHEAT_GUEST_THIRST, nullptr, COLOUR_BLACK, w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(4) + TXTO });
         gfx_draw_string_left(
-            dpi, STR_CHEAT_GUEST_NAUSEA, nullptr, COLOUR_BLACK, w->windowPos.x + XPL(0) + TXTO, w->windowPos.y + YPL(5) + TXTO);
+            dpi, STR_CHEAT_GUEST_NAUSEA, nullptr, COLOUR_BLACK, w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(5) + TXTO });
         gfx_draw_string_left(
-            dpi, STR_CHEAT_GUEST_NAUSEA_TOLERANCE, nullptr, COLOUR_BLACK, w->windowPos.x + XPL(0) + TXTO,
-            w->windowPos.y + YPL(6) + TXTO);
+            dpi, STR_CHEAT_GUEST_NAUSEA_TOLERANCE, nullptr, COLOUR_BLACK,
+            w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(2) + TXTO });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Cheats.cpp:1253]:
```suggestion
            w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(6) + TXTO });
```
Used to be 6

File: src/openrct2-ui/windows/Cheats.cpp [hunk 6]
```diff
@@ -1207,55 +1207,55 @@ static void window_cheats_paint(rct_window* w, rct_drawpixelinfo* dpi)
...
         gfx_draw_string_left(
-            dpi, STR_CHEAT_GUEST_TOILET, nullptr, COLOUR_BLACK, w->windowPos.x + XPL(0) + TXTO, w->windowPos.y + YPL(7) + TXTO);
+            dpi, STR_CHEAT_GUEST_TOILET, nullptr, COLOUR_BLACK, w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(2) + TXTO });
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Cheats.cpp:1255]:
```suggestion
            dpi, STR_CHEAT_GUEST_TOILET, nullptr, COLOUR_BLACK, w->windowPos + ScreenCoordsXY{ XPL(0) + TXTO, YPL(7) + TXTO });
```
Used to be 7

File: src/openrct2-ui/windows/EditorScenarioOptions.cpp [hunk 7]
```diff
@@ -742,55 +742,51 @@ static void window_editor_scenario_options_financial_invalidate(rct_window* w)
  */
 static void window_editor_scenario_options_financial_paint(rct_window* w, rct_drawpixelinfo* dpi)
 {
-    int32_t x, y;
+    ScreenCoordsXY screenCoords;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorScenarioOptions.cpp:745]:
```suggestion
    ScreenCoordsXY screenCoords{};
```
Initialize it, just to be sure

File: src/openrct2-ui/windows/EditorScenarioOptions.cpp [hunk 8]
```diff
@@ -1386,7 +1380,8 @@ static void window_editor_scenario_options_park_invalidate(rct_window* w)
  */
 static void window_editor_scenario_options_park_paint(rct_window* w, rct_drawpixelinfo* dpi)
 {
-    int32_t x, y, arg;
+    int32_t arg;
+    ScreenCoordsXY screenCoords;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorScenarioOptions.cpp:1384]:
```suggestion
    ScreenCoordsXY screenCoords{};
```
Initialize it, just to be sure

File: src/openrct2/drawing/Text.cpp [hunk 9]
```diff
@@ -156,6 +156,12 @@ void gfx_draw_string_left(rct_drawpixelinfo* dpi, rct_string_id format, void* ar
     DrawTextCompat(dpi, x, y, format, args, colour, TextAlignment::LEFT);
```
Comment (tupaschoal) [inline src/openrct2/drawing/Text.cpp:156]: Make this call the new function instead, so we don't duplicate code
```suggestion
    gfx_draw_string_left(dpi, format, args, colour, { x, y});
```

**Comment by TomasZilinek** [comment]
I am sorry about changing the numbers. :(

**Comment by tupaschoal** [comment]
It happens, that's why code review exists :)

File: src/openrct2-ui/input/KeyboardShortcuts.cpp [hunk 10]
```diff
@@ -217,7 +217,7 @@ std::string KeyboardShortcuts::GetShortcutString(int32_t shortcut) const
 
 ScreenCoordsXY KeyboardShortcuts::GetKeyboardMapScroll(const uint8_t* keysState) const
 {
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/input/KeyboardShortcuts.cpp:220]: I know you did it with the best of intentions, but this doesn't relate to the task at hand, so can you please revert?
```suggestion
    ScreenCoordsXY screenCoords;
```

File: src/openrct2-ui/input/MouseInput.cpp [hunk 11]
```diff
@@ -110,7 +110,7 @@ void game_handle_input()
     invalidate_all_windows_after_input();
 
     int32_t state;
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/input/MouseInput.cpp:113]: I know you did it with the best of intentions, but this doesn't relate to the task at hand, so can you please revert?
```suggestion
    ScreenCoordsXY screenCoords;
```

File: src/openrct2-ui/windows/EditorObjectSelection.cpp [hunk 12]
```diff
@@ -1116,7 +1115,7 @@ static void window_editor_object_selection_paint(rct_window* w, rct_drawpixelinf
 static void window_editor_object_selection_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi, int32_t scrollIndex)
 {
     int32_t colour, colour2;
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorObjectSelection.cpp:1118]: Same as said before

File: src/openrct2-ui/windows/Footpath.cpp [hunk 13]
```diff
@@ -596,7 +596,7 @@ static void window_footpath_invalidate(rct_window* w)
  */
 static void window_footpath_paint(rct_window* w, rct_drawpixelinfo* dpi)
 {
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Footpath.cpp:599]: Same

File: src/openrct2-ui/windows/Land.cpp [hunk 14]
```diff
@@ -339,7 +339,7 @@ static void window_land_invalidate(rct_window* w)
  */
 static void window_land_paint(rct_window* w, rct_drawpixelinfo* dpi)
 {
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Land.cpp:342]: Same

File: src/openrct2-ui/windows/Multiplayer.cpp [hunk 15]
```diff
@@ -644,7 +644,7 @@ static void window_multiplayer_players_paint(rct_window* w, rct_drawpixelinfo* d
 
 static void window_multiplayer_players_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi, int32_t scrollIndex)
 {
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Multiplayer.cpp:647]: Same

File: src/openrct2-ui/windows/ObjectLoadError.cpp [hunk 16]
```diff
@@ -578,7 +578,7 @@ static void window_object_load_error_scrollpaint(rct_window* w, rct_drawpixelinf
 
     for (int32_t i = 0; i < w->no_list_items; i++)
     {
-        ScreenCoordsXY screenCoords;
+        ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/ObjectLoadError.cpp:581]: Same

File: src/openrct2-ui/windows/ServerList.cpp [hunk 17]
```diff
@@ -431,7 +431,7 @@ static void window_server_list_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi
 
     int32_t width = w->widgets[WIDX_LIST].right - w->widgets[WIDX_LIST].left;
 
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/ServerList.cpp:434]: Same

File: src/openrct2-ui/windows/TextInput.cpp [hunk 18]
```diff
@@ -252,7 +252,7 @@ static void window_text_input_paint(rct_window* w, rct_drawpixelinfo* dpi)
 {
     window_draw_widgets(w, dpi);
 
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TextInput.cpp:255]: Same

File: src/openrct2-ui/windows/Themes.cpp [hunk 19]
```diff
@@ -855,7 +855,7 @@ void window_themes_paint(rct_window* w, rct_drawpixelinfo* dpi)
  */
 void window_themes_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi, int32_t scrollIndex)
 {
-    ScreenCoordsXY screenCoords;
+    ScreenCoordsXY screenCoords{};
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Themes.cpp:858]: Same

**Comment by TomasZilinek** [comment]
> This looks good now, I just opened issues for places where you touched that shouldn't really.
> 
> As a rule of thumb, try to keep changes in a commit related, don't try to fix other stuff at the same time.

So should I change it back? The unrelated changes?

**Comment by tupaschoal** [comment]
Yes, just the ones I commented last.

Do remember to mark issues as resolved once you fix them, please :)

**Comment by tupaschoal** [comment]
I've squash merged it. Note that I've reworded the commit to be "Part of #XXXXX"  instead of "Fix #XXXXX". The latter would automatically close the issue when I merged, and since we know there are still places to work on, we don't want that :)

**Comment by TomasZilinek** [comment]
> I've squash merged it. Note that I've reworded the commit to be "Part of #XXXXX" instead of "Fix #XXXXX". The latter would automatically close the issue when I merged, and since we know there are still places to work on, we don't want that :)

Oh, I was unaware of an automatic mechanism that reacts to words in commits. Is it ok to word the PRs like: "Fix XXXXX part Y" or is this just with commit messages.

**Comment by duncanspumpkin** [comment]
@TomasZilinek pretty sure its just the commit message. I'm pretty sure the PR title doesn't even link related issues if the reference is in the title so always have the reference in the commit message and also preferably in the PR body text.

**Pull request merged**